### PR TITLE
chain: implement collect dataSource fee from requester

### LINKED ISSUE #756: Collect data source processing fees from data requester
If data source fee is not zero, the data requester should pay for the query.

### PR DESCRIPTION
**Pull request by evilpeach**
Fixes #756

### CODE REVIEW

File: chain/x/zoracle/handler.go [hunk 1]
```diff
@@ -251,6 +251,11 @@ func handleMsgRequestData(ctx sdk.Context, keeper Keeper, msg MsgRequestData) sd
 		return err.Result()
 	}
 
+	err = keeper.PayDataSourcesFee(ctx, id, msg.Sender)
```
Comment (sorawit) [inline chain/x/zoracle/handler.go]: PayDataSourceFees

File: chain/x/zoracle/internal/keeper/request.go [hunk 2]
```diff
@@ -97,6 +97,28 @@ func (k Keeper) ValidateDataSourceCount(ctx sdk.Context, id types.RequestID) sdk
 	return nil
 }
 
+// PayDataSourcesFee send fees to the owner of requested data sources.
```
Comment (sorawit) [inline chain/x/zoracle/internal/keeper/request.go]: sends

File: chain/x/zoracle/internal/keeper/request.go [hunk 3]
```diff
@@ -97,6 +97,28 @@ func (k Keeper) ValidateDataSourceCount(ctx sdk.Context, id types.RequestID) sdk
...
+func (k Keeper) PayDataSourcesFee(ctx sdk.Context, id types.RequestID, sender sdk.AccAddress) sdk.Error {
+	rawDataRequests := k.GetRawDataRequests(ctx, id)
+	for _, rawDataRequest := range rawDataRequests {
+		dataSource, err := k.GetDataSource(ctx, rawDataRequest.DataSourceID)
+		if err != nil {
+			return err
+		}
+
+		if dataSource.Fee.IsZero() {
```
Comment (sorawit) [inline chain/x/zoracle/internal/keeper/request.go:113]: Let's also do nothing if the sender is the same as owner

**Review by sorawit — CHANGES_REQUESTED** [review]

File: chain/x/zoracle/internal/keeper/request.go [hunk 4]
```diff
@@ -97,6 +97,28 @@ func (k Keeper) ValidateDataSourceCount(ctx sdk.Context, id types.RequestID) sdk
 	return nil
 }
 
+// PayDataSourcesFee send fees to the owner of requested data sources.
+func (k Keeper) PayDataSourcesFee(ctx sdk.Context, id types.RequestID, sender sdk.AccAddress) sdk.Error {
```
Comment (taobun) [inline chain/x/zoracle/internal/keeper/request.go]: Can you add a simple test for this function? By set a request. data source and raw data request and make sure the balance of requester is reduced.

**Review by taobun — CHANGES_REQUESTED** [review]

File: chain/x/zoracle/internal/keeper/request.go [hunk 5]
```diff
@@ -97,6 +97,32 @@ func (k Keeper) ValidateDataSourceCount(ctx sdk.Context, id types.RequestID) sdk
 	return nil
 }
 
+// PayDataSourceFees sends fees to the owner of requested data sources.
```
Comment (sorawit) [inline chain/x/zoracle/internal/keeper/request.go]: owners of the requested

**Review by sorawit — CHANGES_REQUESTED** [review]

**Review by sorawit — APPROVED** [review]

**Review by taobun — APPROVED** [review]
LGTM

**Pull request merged**